Home work

Kaftaranov opened the pull request.

Kaftaranov committed:
Project release.

Kaftaranov committed:
Project release.

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,49 @@
+private static final int allowedLoginLength = 20;
+private static final int allowedPasswordLength = 19;
+private static final String allowedSymbols = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890_";
+    public static Boolean userCredentialsValidation(String login, String password, String confirmPassword)
```
можно было бы примитив вернуть, а не обёртку, т.е. boolean. Обычно обёртки возвращают, когда кроме самих значений обёрток метод может вернуть null. У тебя метод null никогда не вернёт, всегда либо true, либо false, поэтому смысла особого в обёртке нет

dmitry-bizin commented on src/WrongLoginException.java:
```diff
@@ -0,0 +1,6 @@
+import java.io.IOException;
+public class WrongLoginException extends IOException {
+    public WrongLoginException(){
```
можно было бы не писать конструктор по умолчанию

dmitry-bizin commented on src/WrongPasswordException.java:
```diff
@@ -0,0 +1,6 @@
+import java.io.IOException;
+public class WrongPasswordException extends IOException {
+    public WrongPasswordException() {
```
можно было бы не писать конструктор по умолчанию

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,49 @@
+            throw new WrongPasswordException ();
+        }
+        if (password.length() > allowedPasswordLength){
+            System.out.println("Maximum length of password is " + allowedPasswordLength + " symbols.");
```
здесь нужно было тоже выкинуть исключение WrongPasswordException

dmitry-bizin replied:
вместо вывода и возврата

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,49 @@
+        }
+            for (int i = 0; i < login.length(); i++) {
+                    if (!allowedSymbols.contains(login.substring(i,i+1))) {
+                        System.out.println("Login should contain latin letters, numbers and underline symbols.");
```
здесь нужно выкинуть исключение WrongLoginException

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,49 @@
+            return false;
+        }
+            for (int i = 0; i < login.length(); i++) {
+                    if (!allowedSymbols.contains(login.substring(i,i+1))) {
```
вместо substring лучше `String.valueOf(login.charAt(i))`

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,49 @@
+                        return false;
+                    }
+                }
+
```
нет проверки символов пароля как для логина ты делаешь

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,49 @@
+       }
+
+    } catch (WrongLoginException e){
+        System.out.println("Maximum length of login is " + allowedLoginLength + " symbols.");
```
вот у тебя разбросана (размазана) логика по исключениям и boolean. Лучше вот как сделать: приватный метод, который делает проверки и выкидывает исключения. Он ничего не возвращает. Публичный метод вызывает этот приватный метод в блоке try и ловит исключения. Если поймал, то возвращает false, иначе возвращает true. Этот публичный метод нужно вызвать в методе main и получить результат - boolean. Если результат true, то проверка прошла, иначе не прошла

Kaftaranov commented:
В задании сказано, что исключение следует выбросить только в случае неверной длины.

Отправлено из приложения "Почта" для Windows

От: Dmitry Bizin
Отправлено: воскресенье, 26 июня 2022 г. в 19:33
Кому: Kaftaranov/ExceptionsHandling
Копия: Timur Kaftaranov; Author
Тема: Re: [Kaftaranov/ExceptionsHandling] Home work (PR #1)

@BizinMitya commented on this pull request.

In src/Main.java:
> +    throws WrongPasswordException, WrongLoginException

+    {

+        if (login.length() > allowedLoginLength) {

+            throw new WrongLoginException();

+        }

+        if (!password.equals(confirmPassword)) {

+            throw new WrongPasswordException ();

+        }

+        if (password.length() > allowedPasswordLength){

+            System.out.println("Maximum length of password is " + allowedPasswordLength + " symbols.");

+            System.out.println("Length of your password is " + password.length() + " symbols.");

+            return false;

+        }

+            for (int i = 0; i < login.length(); i++) {

+                    if (!allowedSymbols.contains(login.substring(i,i+1))) {

+                        System.out.println("Login should contain latin letters, numbers and underline symbols.");

здесь нужно выкинуть исключение WrongLoginException
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Kaftaranov committed:
Check password for allowed symbols added.
Method "substring" replaced by "String.valueOf()".
"Boolean" in method caption corrected to "boolean".
Exceptions for all irregularities in credentials introduced.
Exceptions handling moved to separate boolean method.

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,55 @@
+    public static boolean validationSuccessful() {
+        try {
+            userCredentialsValidation(login, password, confirmPassword);
+        }catch (WrongLoginException loginException){
```
вот эти два catch можно объединить в один, т.к. они обрабатывают исключения одинаково

dmitry-bizin commented:
Да, верно, в задании не всегда нужно выкидывать исключение...Хм, зачем так написали интересно.
